db/Controllers/api.controller.js

```diff
@@ -1,38 +1,50 @@
-const endpoints = require('../../endpoints.json')
-const {readTopics, readArticleId} = require('../models/api.model')
+const endpoints = require("../../endpoints.json");
+const {
+  readTopics,
+  readArticleId,
+  readArticles,
+} = require("../models/api.model");
 
 exports.getApiDocu = (req, res) => {
-    res.status(200).send({endpoints})
-}
+  res.status(200).send({ endpoints });
+};
 
 exports.getTopics = (req, res) => {
-        readTopics().then((topics) => {
-            res.status(200).send({topics})
-        })
-        .catch((err) => {
-            console.error("Error fetching topics:", err);
-            res.status(500).send({msg: "server error"})
-        })
-}
+  readTopics()
+    .then((topics) => {
+      res.status(200).send({ topics });
+    })
+    .catch((err) => {
+      console.error("Error fetching topics:", err);
+      res.status(500).send({ msg: "server error" });
+    });
+};
 
 exports.getArticleId = (req, res) => {
+  const { article_id } = req.params;
 
-    const { article_id } = req.params;
-
-    readArticleId(article_id)
+  readArticleId(article_id)
     .then((article) => {
-        if (!article) {
-            return res.status(404).send({msg: "Article not found"})
-        } else{
-            res.status(200).send({ article })
-        }
-
+      if (!article) {
+        return res.status(404).send({ msg: "Article not found" });
+      } else {
```

Comment:
You should deal with this in the model - check the rows that come back and if they are empty return a promise.reject. 

You can then catch this error in your catch block.